### ISSUE TITLE
URGENT ACTION: Stop using az416426.vo.msecnd.net

### ISSUE THREAD

**Issue by MSNev** [microsoft/ApplicationInsights-JS#2457]
# URGENT ACTION: Stop using az416426.vo.msecnd.net

To avoid any global OUTAGE you MUST change ALL of your CDN usage from “https://az416426.vo.msecnd.net/scripts/..” to our primary CDN endpoint https://js.monitor.azure.com/scripts/...

A change in CDN usage is required to avoid potential outages due to the [bankruptcy of the Verizon Edgio-based CDN provider.](https://www.datacenterdynamics.com/en/news/cdn-provider-edgio-files-for-bankruptcy/) where they have brought forward their timeframe for deactivation of the service.

- CDN Endpoint Change: All references to “az416426.vo.msecnd.net” must be updated to “js.monitor.azure.com” to prevent service disruptions due to the impending disablement of the Verizon Edgio-based CDN endpoint by Mid-January, 2025.
- Migration Challenges: Since the domain “az416426.vo.msecnd.net” is controlled by Verizon, it cannot be redirected or migrated, leading to a high likelihood of temporary or permanent outages.
- Action Required: Organizations must update their references and deploy changes to production before the service is disabled, and ensure that their sites can handle the absence of the 1DS JS SDK from the CDN.

Right now we are expecting that there WILL be either a **temporary or permanent outage** of this domain based on the details we currently have. As we currently have no known way to “migrate” this legacy domain to a different Azure CDN definition.

## What do you need to do?

-	Change ALL references of “az416426.vo.msecnd.net” to “js.monitor.azure.com”
-	Deploy the change to production before Verizon disables the service is disabled (we don’t control this disablement)
-	Ensure that your site can handle when Application Insights JS Sdk cannot be loaded from the CDN without breaking your functionality

## Details

Both the `az416426.vo.msecnd.net` and `js.monitor.azure.com` source their files from the same location as such all files and content is identical regardless of which domain endpoint you use.
The endpoints are currently configured as different CDN definitions and are serviced by different CDN providers, so only the az416426.vo.msecnd.net is being affected by this situation.

## Workarounds

-	Stop using “az416426.vo.msecnd.net” and change to “js.monitor.azure.com”
-	There is currently NO other workaround

We are currently investigating the available options on how we can avoid / migrate / mitigate this situation, but at this point it is **HIGHLY** likely that there will be either a temporary or permanent outage of this domain. As we currently have no known way to “migrate” this domain to a different CDN.

Any updates will be added to this issue.

**Comment by MSNev** [comment]
Why is this occurring?

Please see this public announcement [Azure CDN from Edgio retirement FAQ](https://learn.microsoft.com/en-us/azure/cdn/edgio-retirement-faq) where this domain (`az416426.vo.msecnd.net` is currently an Edgio based CDN instance)

**Comment by baxxos** [comment]
Just to clarify - once this is fixed, the users will have to raise their version of Application Insights JS SDK or else no data will be received in Azure past Jan 15th?

**Comment by MSNev** [comment]
Correct, this issue ONLY affects users that are loading the SDK via the legacy CDN, if you are using the newer SDK Loader (snippet) which defaults to the current CDN endpoint (js.monitor.azure.com) or are using the SDK via an `npm` package embedded in your own bundles / domain then you are unaffected by this issue.

**Comment by MSNev** [comment]
The DNS (CNAME) entry has been redirected, and we appear to be seeing successful traffic being redirected to the new CDN. 

However, the DNS Zone delegation still needs to be migrated, so there is still a risk of an outage...

**Comment by MSNev** [comment]
I've just "re-enabled" `http` access on this legacy endpoint (it had been defaulted to redirect all `http` -> `https`).

It appears that referrers that had been requesting `http` versions where successfully requestion `https` version (as a follow up request from the same referrer)

**Comment by MSNev** [comment]
DNS Delegation has just been migrated and appears to be resolving as expected.
One more final step scheduled for tomorrow (PST) to complete the migration process.

**Comment by MSNev** [comment]
The final mitigation step is now complete, significantly reducing the risk of an outage for this domain due to the Edgio retirement.

However, this domain is now entering a more orderly deprecation phase, with exact timeframes to be determined as we move beyond this unexpected situation.
